kglib/kgcn/pipeline/pipeline.py

```diff
@@ -82,18 +86,28 @@ def pipeline(graphs,
                 node_output_size=node_output_size)
 
     learner = KGCNLearner(kgcn,
-                          num_processing_steps_tr=num_processing_steps_tr,
-                          num_processing_steps_ge=num_processing_steps_ge)
-
-    train_values, test_values, tr_info = learner(tr_input_graphs,
+                          num_processing_steps_tr=num_processing_steps_tr, # These processing steps indicate how many message-passing iterations to do for every training / testing step
+                          num_processing_steps_ge=num_processing_steps_ge,
+                          save_fle=f'{output_dir}/{save_fle}',
+                          reload_fle=f'{output_dir}/{reload_fle}')
+
+    # only test
+    if not Path(output_dir / reload_fle).is_dir() and do_test == True:
+        test_values, tr_info = learner.test(ge_input_graphs,
+                                            ge_target_graphs,
+                                            log_dir=output_dir)
+    # train
+    else:
+        train_values, test_values, tr_info = learner(input_graphs,
```

Comment:
Now that you've added `learner.test(...)` it would be ideal to change this method to be `learner.train(...)`, I don't know if this breaks anything though, I can't remember if it will cause an issue with Sonnet or not, but worth a try. That means renaming `KGCNlearner`'s `__call__` magic method with `train`

Reply:
Sounds logical. 
I adjusted it. We will see what happens.